azure-kusto-data/azure/kusto/data/_models.py

```diff
@@ -99,6 +99,11 @@ def __repr__(self):
         values = [repr(val) for val in self._value_by_name.values()]
         return "KustoResultRow(['{}'], [{}])".format("', '".join(self._value_by_name), ", ".join(values))
 
+    def __eq__(self, other):
+        if len(self) != len(other):
+            return False
+        return repr(self) == repr(other)
```

Comment:
This feels slow and with an overhead of allocations.
Why not compare value_by_name directly?